### PR TITLE
Added envoy tokens endpoint

### PR DESCRIPTION
**Pull request by itzg**
# Resolves

https://jira.rax.io/browse/SALUS-846

# What

Added the public API endpoint for envoy tokens:

https://github.com/Rackspace-Segment-Support/salus-docs-internal/blob/master/proposals/envoy-token-for-cert-retrieval/envoy-token-for-cert-retrieval.md#public-api

### CODE REVIEW

File: public/src/main/java/com/rackspace/salus/telemetry/api/web/EnvoyTokensController.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+  }
+
+  @GetMapping("/tenant/{tenantId}/envoy-tokens/{token}")
+  public ResponseEntity<?> getOne(ProxyExchange<?> proxy,
+                                  @PathVariable String tenantId,
+                                  @PathVariable String token,
+                                  @RequestHeader HttpHeaders headers) {
+    final String backendUri = UriComponentsBuilder
+        .fromUriString(servicesProperties.getAuthServiceUrl())
+        .path("/api/tenant/{tenantId}/envoy-tokens/{token}")
```
Comment (jjbuchan) [inline public/src/main/java/com/rackspace/salus/telemetry/api/web/EnvoyTokensController.java:86]: A thought about this... we probably need to ensure the `token` value is redacted in any of our logs?

Comment (itzg) [reply public/src/main/java/com/rackspace/salus/telemetry/api/web/EnvoyTokensController.java:86]: Oh yeah...I think I'll write up an issue to investigate a logback way to do that generically.

Comment (harageth) [reply public/src/main/java/com/rackspace/salus/telemetry/api/web/EnvoyTokensController.java:86]: Are there other values in the system that we need to redact? mysql/postgres passwords?

Comment (itzg) [reply public/src/main/java/com/rackspace/salus/telemetry/api/web/EnvoyTokensController.java:86]: It didn't know of any others, but does seem like something logback, etc would have solutions for.

Comment (itzg) [reply public/src/main/java/com/rackspace/salus/telemetry/api/web/EnvoyTokensController.java:86]: ...yep, got several search hits and this type of solution looks viable

https://medium.com/@kolapkar.dhaval/mask-sensitive-data-in-logs-7e06496e56c1

Comment (harageth) [reply public/src/main/java/com/rackspace/salus/telemetry/api/web/EnvoyTokensController.java:86]: Seems reasonable. Although do we have certain patterns that we will be able to match against? Is the token value specific enough to catch with a regexp?

Comment (itzg) [reply public/src/main/java/com/rackspace/salus/telemetry/api/web/EnvoyTokensController.java:86]: I created a follow up Jira issue 

https://jira.rax.io/browse/SALUS-864

where I'm also proposing we simply introduce a "database ID" field into the `EnvoyToken` and use that for the REST API path variable.

**Pull request merged**